core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -855,7 +855,7 @@ private[spark] class Executor(
    * Schedules a task to report heartbeat and partial metrics for active tasks to driver.
    */
   private def startDriverHeartbeater(): Unit = {
-    val intervalMs = conf.getTimeAsMs("spark.executor.heartbeatInterval", "10s")
+    val intervalMs = conf.getTimeAsSeconds("spark.executor.heartbeatInterval", "10s") * 1000
```

Comment:
Please, avoid magic constants and use either:
- `DateTimeUtils.MILLIS_PER_SECOND`
- or `TimeUnit.SECONDS.toMillis()`

Reply:
updated to ``TimeUnit.SECONDS.toMillis()`` as ``DateTimeUtils.MILLIS_PER_SECOND`` belongs to catalyst